src/snowflake/cli/plugins/snowpark/commands.py

```diff
@@ -308,31 +314,108 @@ def _deploy_single_object(
+        "--pypi-download flag is deprecated. Snowpark build command"
+        " always tries to download non-Anaconda packages from PyPi."
+    ),
+)
+
+
+def _write_requirements_file(file_path: SecurePath, requirements: List[Requirement]):
+    log.info("Writing requirements into file %s", file_path.path)
+    file_path.write_text("\n".join(req.line for req in requirements))
+
+
 @app.command("build")
 @with_project_definition("snowpark")
 def build(
-    pypi_download: YesNoAsk = PyPiDownloadOption,
-    check_anaconda_for_pypi_deps: bool = CheckAnacondaForPyPiDependencies,
-    package_native_libraries: YesNoAsk = PackageNativeLibrariesOption,
+    ignore_anaconda: bool = IgnoreAnacondaOption,
+    allow_shared_libraries: bool = AllowSharedLibrariesOption,
+    index_url: Optional[str] = IndexUrlOption,
+    skip_version_check: bool = SkipVersionCheckOption,
+    deprecated_package_native_libraries: YesNoAsk = deprecated_allow_native_libraries_option(
+        "--package-native-libraries"
+    ),
+    deprecated_check_anaconda_for_pypi_deps: bool = DeprecatedCheckAnacondaForPyPiDependencies,
+    _deprecated_pypi_download: YesNoAsk = deprecated_pypi_download_option,
     **options,
 ) -> CommandResult:
     """
     Builds the Snowpark project as a `.zip` archive that can be used by `deploy` command.
     The archive is built using only the `src` directory specified in the project file.
     """
-    paths = SnowparkPackagePaths.for_snowpark_project(
+    if not deprecated_check_anaconda_for_pypi_deps:
+        ignore_anaconda = True
+    # TODO: yes/no/ask logic should be removed in 3.0
+    allow_shared_libraries_yesnoask = {
+        True: YesNoAsk.YES,
+        False: YesNoAsk.NO,
+    }[allow_shared_libraries]
+    if deprecated_package_native_libraries != YesNoAsk.NO:
+        allow_shared_libraries_yesnoask = deprecated_package_native_libraries
+
+    snowpark_paths = SnowparkPackagePaths.for_snowpark_project(
         project_root=SecurePath(cli_context.project_root),
         snowpark_project_definition=cli_context.project_definition,
     )
-    log.info("Building package using sources from: %s", paths.source.path)
+    log.info("Building package using sources from: %s", snowpark_paths.source.path)
+
+    with SecurePath.temporary_directory() as packages_dir:
+        if snowpark_paths.defined_requirements_file.exists():
+            log.info("Resolving any requirements from requirements.txt...")
+            requirements = package_utils.parse_requirements(
+                requirements_file=snowpark_paths.defined_requirements_file,
+            )
+
+            download_result = package_utils.download_unavailable_packages(
+                requirements=requirements,
+                target_dir=packages_dir,
+                ignore_anaconda=ignore_anaconda,
+                skip_version_check=skip_version_check,
+                pip_index_url=index_url,
+                allow_shared_libraries=allow_shared_libraries_yesnoask,
```

Comment:
It seems that native libs checks happens later on, why download requires the argument?

Reply:
Propably a leftover after merge - nice catch!